contracts/rewarders/BarRewarderPerSec.sol

```diff
@@ -0,0 +1,119 @@
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import "../boringcrypto/BoringOwnable.sol";
+import "../libraries/SafeERC20.sol";
+
+interface IBarV2 {
+    function totalSupply() external view returns (uint256);
+
+    function balanceOf(address account) external view returns (uint256);
+
+    function entryFee() external view returns (uint256);
+}
+
+/**
+ * This is a sample contract to be used in the JoeBarV2 to reward moJoe holder with JOE.
+ *
+ * It assumes no minting rights, so requires a set amount of YOUR_TOKEN to be transferred to this contract prior.
+ * E.g. say you've allocated 100,000 XYZ to the JOE-XYZ farm over 30 days. Then you would need to transfer
+ * 100,000 XYZ and set the block reward accordingly so it's fully distributed after 30 days.
+ *
+ */
+contract BarRewarderPerSec is BoringOwnable, ReentrancyGuard {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    IERC20 public immutable joe;
+    IBarV2 public immutable bar;
+    uint256 public apr;
+
+    uint256 private secondsPerYear = 365 days;
+
+    uint256 public lastRewardTimestamp;
+    uint256 public tokenPerSec;
+
+    uint256 public unpaidRewards;
+
+    event ClaimReward(uint256 amount);
+    event AprUpdated(uint256 oldRate, uint256 newRate);
+
+    modifier onlyBar() {
+        require(msg.sender == address(bar), "onlyBar: only JoeBar can call this function");
+        _;
+    }
+
+    constructor(
+        IERC20 _joe,
+        IBarV2 _bar,
+        uint256 _apr
+    ) public {
+        require(Address.isContract(address(_joe)), "constructor: reward token must be a valid contract");
+        require(Address.isContract(address(_bar)), "constructor: LP token must be a valid contract");
+
+        joe = _joe;
+        bar = _bar;
+        apr = _apr;
+        lastRewardTimestamp = block.timestamp;
+        tokenPerSec = 0;
+    }
+
+    /// @notice Update reward variables.
+    function update() public {
+        if (block.timestamp > lastRewardTimestamp) {
+            uint256 barSupply = bar.totalSupply();
+            if (barSupply > 0) {
+                uint256 timeElapsed = block.timestamp.sub(lastRewardTimestamp);
+                tokenPerSec = barSupply.mul(apr).div(10000).div(secondsPerYear);
+                unpaidRewards = unpaidRewards.add(timeElapsed.mul(tokenPerSec));
+            }
+
+            lastRewardTimestamp = block.timestamp;
+        }
+    }
+
+    /// @notice Sets the APR.
+    /// @param _apr The new APR
+    function setApr(uint256 _apr) external onlyOwner {
+        uint256 oldApr = apr;
+        apr = _apr; // in basis points aka parts per 10,000 so 5000 is 50%, apr of 50%.
+        //        apr = _apr.mul(10_000).div(10_000 - IBarV2.entryFee()); // if added then when bar updates its fees
+        //                                                          this needs to be called to update the actual JOE apr
```

Comment:
If this is commented out code, then please remove

Reply:
I actually don't know what apr do we want to use.
if we input apr = 15%,
- does it mean 15% on moJoe (95% of joe sent) (actual)
- or does it mean 15% on actual Joe sent ? (commented)

Reply:
Hold on this for now; we're still discussing the correct mechanics of this in discord